HW1_Skeleton/AStarSearcher.java

```diff
@@ -36,24 +36,67 @@ public boolean search() {
 		// TODO initialize the root state and add
 		// to frontier list
 		// ...
-		Square start = maze.getPlayerSquare();
-		Square goal = maze.getGoalSquare();
-		int hvalue = (int)Math.sqrt(((start.X - goal.X)^2)+ ((start.Y - goal.Y)^2));
+		Square startSquare = maze.getPlayerSquare();
+		Square goalSquare = maze.getGoalSquare();
+		int rootFValue = (int) Math.sqrt(((startSquare.X - goalSquare.X) ^ 2) + ((startSquare.Y - goalSquare.Y) ^ 2));
 
-		State root = new State(start,null,0,0);
+		State root = new State(startSquare, null, 0, 0);
+		StateFValuePair rootPair = new StateFValuePair(root, rootFValue);
+
+		frontier.add(rootPair);
 
 		while (!frontier.isEmpty()) {
 			// TODO return true if a solution has been found
 			// TODO maintain the cost, noOfNodesExpanded (a.k.a. noOfNodesExplored),
 			// maxDepthSearched, maxSizeOfFrontier during
 			// the search
 			// TODO update the maze if a solution found
+			StateFValuePair currentPair = frontier.poll();
+			State currentState = currentPair.getState();
+			Square currentSquare = currentState.getSquare();
+
+			// Marked current square explored:
+			int currentX = currentSquare.X;
+			int currentY = currentSquare.Y;
+			explored[currentX][currentY] = true;
+			maze.setOneSquare(currentSquare,'*');
+
+			if (currentState.isGoal(maze)){
+				return true;
+			} else {
+
+				ArrayList<State> currentSuccessors = currentState.getSuccessors(explored, maze);
+				this.maxDepthSearched++;
+				this.cost++;
+
+				for (int j = 0; j < currentSuccessors.size(); j++) {
+					State successorState = currentSuccessors.get(j);
+					Square successorSquare = successorState.getSquare();
+					int successorX = successorSquare.X;
+					int successorY = successorSquare.Y;
+					if (!explored[successorX][successorY]) {
```

Comment:
You don't need to compare the `explored` here, the `explored` is already taken care of in `getSuccessors`.